src/main/scala/rise/core/IsClosedForm.scala

```diff
@@ -69,25 +69,37 @@ object IsClosedForm {
 
     override def natToData: NatToData => Pair[NatToData] = {
       case NatToDataLambda(x, e) =>
-        for { p <- this.copy(boundT = boundT + NatKind.IDWrapper(x)).`type`(e)}
-          yield (p._1, NatToDataLambda(x, e))
+//        for { p <- this.copy(boundT = boundT + NatKind.IDWrapper(x)).`type`(e)}
+//          yield (p._1, NatToDataLambda(x, e))
+        val pp: Pure[((OrderedSet[Identifier], OrderedSet[Kind.Identifier]), DataType)] =
+          this.copy(boundT = boundT + NatKind.IDWrapper(x)).`type`(e)
+        pp.map(p => (p._1, NatToDataLambda(x, e)))
```

Comment:
is the `for` syntax not supported anymore?

Reply:
No, of course it is, but here the system got confused somehow (and I with it ...).
I think that maybe the type inference has changed, and therefore the type annotation helps here, but I am not sure.
The explicit version with the `map` should be the correct one - from my point of view at least.